### PR TITLE
Azure Identity add more test coverage

### PR DESCRIPTION
**Pull request by catalinaperalta**
This PR increases test coverage for untested paths in Azure Identity. Increases package test coverage to 81.3%.

### CODE REVIEW

File: sdk/azidentity/chained_token_credential_test.go [hunk 1]
```diff
@@ -35,6 +35,21 @@ func TestChainedTokenCredential_InstantiateSuccess(t *testing.T) {
...
 	}
+}
+
+func TestChainedTokenCredential_InstantiateFailure(t *testing.T) {
+	secCred, err := NewClientSecretCredential(tenantID, clientID, secret, nil)
+	if err != nil {
+		t.Fatalf("Unable to create credential. Received: %v", err)
+	}
+	_, err = NewChainedTokenCredential(secCred, nil)
+	if err == nil {
```
Comment (jhendrixMSFT) [inline sdk/azidentity/chained_token_credential_test.go:46]: `errors.As()` to verify the error type?

File: sdk/azidentity/azure_cli_credential_test.go [hunk 2]
```diff
@@ -53,3 +55,24 @@ func TestAzureCLICredential_GetTokenInvalidToken(t *testing.T) {
...
+	pipeline := azcore.NewPipeline(
+		srv,
+		azcore.NewTelemetryPolicy(azcore.TelemetryOptions{}),
+		azcore.NewUniqueRequestIDPolicy(),
+		azcore.NewRetryPolicy(nil),
+		cred.AuthenticationPolicy(azcore.AuthenticationPolicyOptions{Options: azcore.TokenRequestOptions{Scopes: []string{scope}}}),
+		azcore.NewRequestLogPolicy(azcore.RequestLogOptions{}))
+	_, err = pipeline.Do(context.Background(), azcore.NewRequest(http.MethodGet, srv.URL()))
+	if err != nil {
+		t.Fatalf("Expected nil error but received one")
```
Comment (jhendrixMSFT) [inline sdk/azidentity/azure_cli_credential_test.go:76]: `t.Fatal(err)`

File: sdk/azidentity/chained_token_credential_test.go (hunk 1 above)
Comment (catalinaperalta) [reply sdk/azidentity/chained_token_credential_test.go:46]: Done

File: sdk/azidentity/azure_cli_credential_test.go (hunk 2 above)
Comment (catalinaperalta) [reply sdk/azidentity/azure_cli_credential_test.go]: Done

**Pull request merged**